metrics/prometheus.go

```diff
@@ -0,0 +1,155 @@
+package metrics
+
+import (
+	"net/http"
+	"os"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "traefik"
+)
+
+// TODO: #156 - More detailed metrics
+type entrypointMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type providerMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type frontendMetrics struct {
+}
+
+// TODO: #156 - More detailed metrics
+type backendMetrics struct {
+}
+
+type globalMetrics struct {
+	UpTime                    prometheus.Gauge
+	RequestResponseTimeTotal  prometheus.Gauge
```

Comment:
In the case of proxying third-party counters, the metric type should still be a counter. But when proxying metrics like this where the authoritative state of the metric is not managed by the Prometheus client library, it is recommended to implement the client library's `Collector` interface and from the `Collect` method, construct throwaway `ConstMetrics` with the proxied values at every `/metrics` scrape.

Take a look at how the memcached_exporter for Prometheus does this, for example: https://github.com/prometheus/memcached_exporter/blob/master/main.go

Of course, the better eventual solution is direct instrumentation of requests such that a long-lived Prometheus counter is incremented whenever a request happens.